drivers/sensor/dps310/dps310.c

```diff
@@ -0,0 +1,755 @@
+	k_msleep(IFX_DPS310_PSR_MEAS_TIME);
+
+	if (!poll_rdy(data, config, IFX_DPS310_REG_ADDR_MEAS_CFG,
+		      IFX_DPS310_REG_ADDR_MEAS_CFG_PRS_RDY)) {
+		LOG_DBG("Poll timeout for pressure");
+		return false;
+	}
+
+	return true;
+}
+
+/*
+ * function to fix a hardware problem on some devices
+ * you have this bug if you measure around 60°C when temperature is around 20°C
+ * call dps310_hw_bug_fix() directly in the init() function to fix this issue
+ */
+static void dps310_hw_bug_fix(struct dps310_data *data,
+			      const struct dps310_cfg *config)
+{
+	/* setup the necessary 5 sequences to fix the hw bug */
+	static const u8_t hw_bug_fix_sequence[HW_BUG_FIX_SEQUENCE_LEN][2] = {
+		/*
+		 * First write valid signature on 0x0e and 0x0f
+		 * to unlock address 0x62
+		 */
+		{ 0x0E, 0xA5 },
+		{ 0x0F, 0x96 },
+		/* Then update high gain value for Temperature */
+		{ 0x62, 0x02 },
+		/* Finally lock back the location 0x62 */
+		{ 0x0E, 0x00 },
+		{ 0x0F, 0x00 }
+	};
+
+	/* execute sequence for hw bug fix */
+	for (int i = 0; i < HW_BUG_FIX_SEQUENCE_LEN; i++) {
+		int res = i2c_write(data->i2c_master, hw_bug_fix_sequence[i], 2,
+				    config->i2c_addr);
+		if (res < 0) {
+			LOG_WRN("I2C error: %d", res);
+			return;
+		}
+	}
+}
+
+/*
+ * Scale and compensate the raw temperature measurement value to micro °C
+ * The formula is based on the Chapter 4.9.2 in the datasheet and was
+ * modified to need only integer arithmetic.
+ */
+static void dps310_scale_temperature(s32_t tmp_raw, struct dps310_data *data)
+{
+	const struct dps310_cal_coeff *comp = &data->comp;
+
+	/* first term, rescaled to micro °C */
+	s32_t tmp_p0 = (1000000 / 2) * comp->c0;
+
+	/* second term, rescaled to mirco °C */
+	s32_t tmp_p1 =
+		(((s64_t)1000000) * comp->c1 * tmp_raw) / IFX_DPS310_SF_TMP;
+
+	/* calculate the temperature corresponding to the datasheet */
+	s32_t tmp_final = tmp_p0 + tmp_p1; /* value is in micro °C */
+
+	/* store calculated value */
+	data->tmp_val1 = tmp_final / 1000000;
+	data->tmp_val2 = tmp_final % 1000000;
+}
+
+/*
+ * Scale and temperature compensate the raw pressure measurement value to
+ * Kilopascal. The formula is based on the Chapter 4.9.1 in the datasheet.
+ */
+static void dps310_scale_pressure(s32_t tmp_raw, s32_t psr_raw,
+				  struct dps310_data *data)
+{
+	const struct dps310_cal_coeff *comp = &data->comp;
+
+	float psr = ((float)psr_raw) / IFX_DPS310_SF_PSR;
+	float tmp = ((float)tmp_raw) / IFX_DPS310_SF_TMP;
+
+	/* scale according to formula from datasheet */
+	float psr_final = comp->c00;
+
+	psr_final += psr * (comp->c10 + psr * (comp->c20 + psr * comp->c30));
+	psr_final += tmp * comp->c01;
+	psr_final += tmp * psr * (comp->c11 + psr * comp->c21);
+
+	/* rescale from Pascal to Kilopascal */
+	psr_final /= 1000;
+
+	/* store calculated value */
+	data->psr_val1 = psr_final;
+	data->psr_val2 = (psr_final - data->psr_val1) * 1000000;
+}
+
+/* Convert the raw sensor data to s32_t */
+static s32_t raw_to_int24(const u8_t raw[3])
```

Comment:
use `sys_le24_to_cpu` instead

Reply:
Replaced with the `sys_get_be*` functions instead, because they fit better, additionally the comment about the values being little endian was wrong, they are big endian.